src/components/Map/map.js

```diff
@@ -281,17 +286,49 @@ class Map extends Component {
           const mapZoom = this.map.getZoom();
           this.map.easeTo({
             center: features[0].geometry.coordinates,
-            zoom: mapZoom >= zoom ? mapZoom + 1 : zoom 
+            zoom: mapZoom >= zoom ? mapZoom + 1 : zoom
           });
         });
     });
   };
 
+  addClusterList = (clusterCenter, list) => {
+    ReactDOM.render(<ClusterList list={list} />, document.getElementById("clusterList"));
+  }
+
+  addClusterMouseOverHandlerToMapLayer = clusterName => {
+    const clusterEl = document.createElement("div");
+    clusterEl.id = "clusterList";
+
+    const clusterListMarker = new mapboxgl.Marker({
+      element: clusterEl,
+      anchor: "bottom",
+      offset: [0, -15],
+    });
+
+    this.map.on("mouseover", clusterName, e => {
```

Comment:
I think these listeners should use mouseenter/mouseleave instead, since those are only fired when you hover over specific layers: https://docs.mapbox.com/mapbox-gl-js/api/#map.event:mouseenter